app/database/user_store.go

```diff
@@ -1,6 +1,102 @@
 package database
 
+import (
+	"fmt"
+)
+
 // UserStore implements database operations on `users`
 type UserStore struct {
 	*DataStore
 }
+
+const deleteWithTrapsBatchSize = 1000
+
+// DeleteTemporaryWithTraps deletes temporary users and all their stuff
+func (s *UserStore) DeleteTemporaryWithTraps() (err error) {
+	defer recoverPanics(&err)
+
+	s.executeBatchesInTransactions(func(store *DataStore) int {
+		userScope := store.Users().
+			Joins("LEFT JOIN sessions ON sessions.idUser = users.ID AND sessions.sExpirationDate > NOW()").
+			Where("sessions.idUser IS NULL").Where("tempUser = 1")
```

Comment:
The query with 'OR' doesn’t work :disappointed: When a user has two sessions (one active and one expired), the query just ignores the active session and the user gets deleted just because he has an expired session.

Reply:
fixed `sessions.sExpirationDate > NOW()`